bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/ValueEncoder.java

```diff
@@ -228,7 +228,14 @@ private static String handleNumericTypes(String dptId, String mainNumber, DPT dp
             }
 
             if (unit != null) {
-                QuantityType<?> converted = ((QuantityType<?>) value).toUnit(unit);
+                QuantityType<?> converted = null;
+                if ("K".equals(unit) || "°C".equals(unit)) {
```

Comment:
Mirek <=> Kelvin conversions make sense. But I am not so sure about Celsius. And if you do want to process Celsius then you probably need to think about Fahrenheit as well. Or ??

Reply:
`unit`  is derived from the DPT type, stored in DPT_UNIT_MAP. There should be only "K" and "°C", those are the units used in KNX.
°C does not make so much sense for color temperatures, but the conversion is handled, according to the tests.